File: library/src/test/java/de/westnordost/osmnames/MapEntry.java
package de.westnordost.osmnames;

import java.util.HashMap;
import java.util.Map;

class MapEntry
{
	private String key, value;
	private MapEntry(String key, String value)
	{
		this.key = key;
		this.value = value;
	}

	static MapEntry tag(String key, String value) { return new MapEntry(key, value); }

	static Map<String, String> mapOf(MapEntry... items)
	{
		Map<String, String> result = new HashMap<>();
		for (MapEntry item : items) result.put(item.key, item.value);
		return result;
	}
}

File: library/src/commonTest/kotlin/osmfeatures/MapEntry.kt
package osmfeatures

internal class MapEntry private constructor(private val key: String, private val value: String) {
    companion object {
        fun tag(key: String, value: String): MapEntry {
            return MapEntry(key, value)
        }

        fun mapOf(vararg items: MapEntry): Map<String, String> {
            val result = hashMapOf<String, String>()
            for (item in items) result[item.key] = item.value
            return result
        }
    }
}
